src/mlpack/methods/ann/ffn.hpp

```diff
@@ -215,6 +217,15 @@ class FFN
   void Forward(arma::mat&& input);
 
   /**
+   * Prepare the network for the given training data.
+   * This function won't actually trigger training process.
+   *
+   * @param predictors Input training variables.
+   * @param responses Outputs results from input training variables.
+   */
+  void SetTrainingData(const arma::mat& predictors, const arma::mat& responses);
```

Comment:
It's a common part of `Gradient()` and `Train()`, I marked it as private. Without it we have to do copy-paste for some code snippet. I'm not sure which is preferred. If we decide to keep it, maybe we can change the name of this function. I don't think current name is so elegant but I can't come up with a better one..